src/net_processing.cpp

```diff
@@ -398,6 +407,9 @@ struct CNodeState {
     //! Whether this peer is a manual connection
     bool m_is_manual_connection;
 
+    //! A rolling bloom filter of all announced tx CInvs to this peer.
+    CRollingBloomFilter m_recently_announced_invs = CRollingBloomFilter{INVENTORY_MAX_RECENT_RELAY, 0.000001};
```

Comment:
I think this naturally belongs in the `TxRelay` struct, along with the rest of the per-peer transaction relay data. It's a bit unfortunate that it's defined in net.h, but it'd mean you don't need to lock cs_main to access it.

Reply:
I agree, but at the same time TxRelay really belongs in net_processing...

I'd rather not move it to net, because that would mean also moving the INVENTORY_MAX_RECENT_RELAY to net.

Reply:
Makes sense. I'm hoping to move all remaining application layer stuff from net to net_processing soon: https://github.com/bitcoin/bitcoin/issues/19398. This can be incorporated into TxRelay when that happens.